pandas/_libs/tslibs/parsing.pyx

```diff
@@ -421,6 +421,10 @@ cdef parse_datetime_string_with_reso(
             parsed = datetime(
                 dts.year, dts.month, dts.day, dts.hour, dts.min, dts.sec, dts.us
             )
+            if out_bestunit == NPY_DATETIMEUNIT.NPY_FR_ns:
```

Comment:
How does this impact the period construction? Feels a bit strange to re-assign this variable